library/Icinga/Web/RememberMe.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/* Icinga Web 2 | (c) 2020 Icinga Development Team | GPLv2+ */
+
+namespace Icinga\Web;
+
+use Icinga\Application\Config;
+use Icinga\Authentication\Auth;
+use Icinga\Crypt\RSA;
+use Icinga\Rememberme\Common\Database;
+use Icinga\User;
+use ipl\Sql\Select;
+use UnexpectedValueException;
+
+class RememberMe
+{
+    use Database;
+
+    /**
+     * Constant cookie
+     */
+    const COOKIE = 'remember-me';
+
+    /**
+     * @var string
+     */
+    protected $encryptedPassword;
+
+    /**
+     * @var string
+     */
+    protected $username;
+
+    /**
+     * @var object
+     */
+    protected $rsa;
+
+    /**
+     * Check if cookie is set
+     *
+     * @return bool
+     *
+     */
+    public static function hasCookie()
+    {
+        return isset($_COOKIE[static::COOKIE]);
+    }
+
+    /**
+     * Get cookie values
+     *
+     * @return void|static
+     */
+    public static function fromCookie()
+    {
+        $data = explode('|', $_COOKIE[static::COOKIE]);
+        $publicKey = base64_decode(array_pop($data));
+
+        $select = (new Select())
+            ->from('rememberme')
+            ->columns('*')
+            ->where(['public_key = ?' => $publicKey]);
+
+        $rememberMe = new static();
+        $dbData = $rememberMe->getDb()->select($select)->fetch();
```

Comment:
Why are you looping over the row and create a new variable here? You may use the result set directly, i.e.

```php
$rs = $rememberMe->getDb()->select($select)->fetch();

$rememberMe->rsa = (new RSA())->loadKey($rs->private_key, $publicKey);
```

Note that we define the following in our database trait

```
PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_OBJ
```

That means, that every fetched row is an instance of `stdClass` with the columns as public properties.

`PDO::FETCH_ASSOC` on the other hand would create us an associative array for every and `PDO::FETCH_BOTH` allows both property and array access.

There is also one thing missing here. Imagine that the database entry is missing. Regardless of how it came to that.

Please have a look what `fetch()` returns in this case and what we should do if that happens.